crypto/include/internal/ec_int.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Internal EC functions for other submodules: not for application use */
+
+#ifndef HEADER_OSSL_EC_INTERNAL_H
+# define HEADER_OSSL_EC_INTERNAL_H
+# include <openssl/opensslconf.h>
+
+# ifndef OPENSSL_NO_EC
+
+#  include <openssl/ec.h>
+
+/*-
+ * Computes the multiplicative inverse of x in the range
+ * [1,EC_GROUP::order).
+ *
+ *         res := x^(-1) (mod EC_GROUP::order)
+ *
+ * - When the EC group order is even, this function returns an error.
+ * - When the EC group order is otherwise composite, the correctness
+ *   of the output is not guaranteed.
+ * - When x is outside the range [1, EC_GROUP::order), the correctness
+ *   of the output is not guaranteed.
```

Comment:
We should probably state what the return values are

Reply:
I'll address return values as soon as I get feedback from @bbbrumley on https://github.com/openssl/openssl/pull/6521#discussion_r197213329.

He has a better understanding than me about the non-default paths when `EC_METHOD`s provide their own specific implementation of this functionality.

What I wrote here (or the alternative text) is definitely correct for the new default path, but given that the specialized versions have been written without documentation I don't want to make wrong assumptions: he looked at that code more than I did!

Reply:
Latest revision explicitly states the return values and the caveats on reported success.